File: node-api/src/services/aboutService.js
import * as repo from "../repository/aboutRepository.js";

export async function addAbout(about) {
  if (!about) {
    throw new Error("About data is required");
  }

  if (!about.presentationText || about.presentationText.trim() === "") {
    throw new Error("Presentation text is required");
  }

  if (about.imageUrl && typeof about.imageUrl !== "string") {
    throw new Error("Invalid image url");
  }

  let id = await repo.saveAbout(about);

  return id;
}

export async function getAbout(filter) {
  let result = await repo.getAbout(filter);
  return result;
}

export async function updateAbout(id, about) {
  if (!id || isNaN(id) || id <= 0) {
    throw new Error("Invalid id");
  }

  if (!about) {
    throw new Error("About data is required");
  }

  if (!about.presentationText || about.presentationText.trim() === "") {
    throw new Error("Presentation text is required");
  }

  return await repo.saveAbout(about); // Save/Update
}

export async function deleteAbout(id) {
  let lines = await repo.deleteAbout(id);
  return lines;
}

export async function getAboutById(id) {
  let line = await repo.getAboutById(id);
  return line;
}

File: spring-api/projeto-fotografico/projeto-fotografico/src/main/java/com/sp/senac/projeto_fotografico/services/AboutService.java
package com.sp.senac.projeto_fotografico.services;

import com.sp.senac.projeto_fotografico.models.About;
import com.sp.senac.projeto_fotografico.repositories.AboutRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.Optional;

@Service
public class AboutService {

    @Autowired
    private AboutRepository repository;

    public Optional<About> getAbout() {
        return repository.findAll().stream().findFirst();
    }

    public Optional<About> getAboutById(int id) {
        return repository.findById(id);
    }

    public About update(int id, String presentationText, String imageUrl) {
        if (presentationText == null || presentationText.isBlank()) {
            throw new IllegalArgumentException("Presentation text is required");
        }

        About about = repository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("About not found"));

        about.setPresentationText(presentationText);
        about.setImageUrl(imageUrl);

        return repository.save(about);
    }

    public void delete(int id) {
        if (!repository.existsById(id)) {
            throw new IllegalArgumentException("About not found");
        }
        repository.deleteById(id);
    }
}
